gnomad/utils/constraint.py

```diff
@@ -700,3 +707,177 @@ def get_all_pop_lengths(
+            `vep_annotation_expr`, PolyPhen annotation from 'polyphen_prediction' in
+            `vep_annotation_expr`, or "None" if neither is defined
+        - gene - 'gene_symbol' annotation inside `vep_annotation_expr`
+        - coverage - exome coverage if `coverage_expr` is specified
+        - transcript - id from 'transcript_id' in `vep_annotation_expr` (added when
+            `include_transcript_group` is True)
+        - canonical from `vep_annotation_expr` (added when `include_canonical_group` is
+            True)
+
+    .. note::
+        This function expects that the following fields are present in
+        `vep_annotation_expr`:
+        - lof
+        - polyphen_prediction
+        - most_severe_consequence
+        - gene_symbol
+        - transcript_id (if `include_transcript_group` is True)
+        - canonical (if `include_canonical_group` is True)
+
+    :param vep_annotation_expr: StructExpression of VEP annotation.
+    :param coverage_expr: Optional Int32Expression of exome coverage. Default is None.
+    :param include_transcript_group: Whether to include the transcript annotation in the
+        groupings. Default is True.
+    :param include_canonical_group: Whether to include canonical annotation in the
+        groupings. Default is True.
+    :return: A dictionary with keys as annotation names and values as actual
+        annotations.
+    """
+    lof_expr = vep_annotation_expr.lof
+    polyphen_prediction_expr = vep_annotation_expr.polyphen_prediction
+
+    # Create constraint annotations to be used for groupings.
+    groupings = {
+        "annotation": vep_annotation_expr.most_severe_consequence,
+        "modifier": hl.coalesce(lof_expr, polyphen_prediction_expr, "None"),
+        "gene": vep_annotation_expr.gene_symbol,
+    }
+    if coverage_expr is not None:
+        groupings["coverage"] = coverage_expr
+
+    # Add 'transcript' and 'canonical' annotation if requested.
+    if include_transcript_group:
+        groupings["transcript"] = vep_annotation_expr.transcript_id
+    if include_canonical_group:
+        groupings["canonical"] = hl.or_else(vep_annotation_expr.canonical == 1, False)
+
+    return groupings
+
+
+def annotate_exploded_vep_for_constraint_groupings(
+    ht: hl.Table, vep_annotation: str = "transcript_consequences"
+) -> Tuple[Union[hl.Table, hl.MatrixTable], Tuple[str]]:
+    """
+    Annotate Table with annotations used for constraint groupings.
+
+    Function explodes the specified VEP annotation (`vep_annotation`) and adds the following
+    annotations:
+    - annotation -'most_severe_consequence' annotation in `vep_annotation`
+    - modifier - classic lof annotation from 'lof' annotation in
+        `vep_annotation_expr`, LOFTEE annotation from 'lof' annotation in
+        `vep_annotation_expr`, PolyPhen annotation from 'polyphen_prediction' in
+        `vep_annotation_expr`, or "None" if neither is defined
```

Comment:
```suggestion
        `vep_annotation`, LOFTEE annotation from 'lof' annotation in
        `vep_annotation`, PolyPhen annotation from 'polyphen_prediction' in
        `vep_annotation`, or "None" if neither is defined
```

Reply:
I remember there was a comment from last round of review:

> Or "None" if neither is defined. Also specify that the first two are pulled from the 'lof' annotation on vep_annotation_expr and the last is from the 'polyphen_prediction' annotation on vep_annotation_expr
> 
> vep_annotation no longer exists

I made the change and just want to double check it is desired!

Reply:
Yup, that change looks good to me. Thanks!